src/contracts/middleware/BLSPubkeyRegistry.sol

```diff
@@ -0,0 +1,213 @@
+    }
+
+    constructor(
+        IRegistryCoordinator _registryCoordinator,
+        IBLSPublicKeyCompendium _pubkeyCompendium
+    ){
+        registryCoordinator = _registryCoordinator;
+        pubkeyCompendium = _pubkeyCompendium;
+    }
+
+    /**
+     * @notice Registers the `operator`'s pubkey for the specified `quorumNumbers`.
+     * @param operator The address of the operator to register.
+     * @param quorumNumbers The quorum numbers the operator is registering for, where each byte is an 8 bit integer quorumNumber.
+     * @param pubkey The operator's BLS public key.
+     * @dev access restricted to the RegistryCoordinator
+     * @dev Preconditions:
+     *         1) `quorumNumbers` has no duplicates
+     *         2) `quorumNumbers.length` != 0
+     *         3) `quorumNumbers` is ordered in ascending order
+     *         4) the operator is not already registered
+     */
+    function registerOperator(address operator, uint8[] memory quorumNumbers, BN254.G1Point memory pubkey) external onlyRegistryCoordinator returns(bytes32){
+        // calculate hash of the operator's pubkey
+        bytes32 pubkeyHash = BN254.hashG1Point(pubkey);
+        require(pubkeyHash != ZERO_PK_HASH, "BLSPubkeyRegistry.registerOperator: cannot register zero pubkey");
+        require(quorumNumbers.length > 0, "BLSPubkeyRegistry.registerOperator: must register for at least one quorum");
+        // ensure that the operator owns their public key by referencing the BLSPubkeyCompendium
+        require(pubkeyCompendium.pubkeyHashToOperator(pubkeyHash) == operator,"BLSPubkeyRegistry.registerOperator: operator does not own pubkey");
+        // update each quorum's aggregate pubkey
+        _processQuorumApkUpdate(quorumNumbers, pubkey);
+        // update the global aggregate pubkey
+        _processGlobalApkUpdate(pubkey);
+        // emit event so offchain actors can update their state
+        emit PubkeyAdded(operator, pubkey);
+        return pubkeyHash;
+    }
+
+    /**
+     * @notice Deregisters the `operator`'s pubkey for the specified `quorumNumbers`.
+     * @param operator The address of the operator to deregister.
+     * @param quorumNumbers The quourm numbers the operator is deregistering from, where each byte is an 8 bit integer quorumNumber.
+     * @param pubkey The public key of the operator.
+     * @dev access restricted to the RegistryCoordinator
+     * @dev Preconditions:
+     *         1) `quorumNumbers` has no duplicates
+     *         2) `quorumNumbers.length` != 0
+     *         3) `quorumNumbers` is ordered in ascending order
+     *         4) the operator is not already deregistered
+     *         5) `quorumNumbers` is the same as the parameter use when registering
+     *         6) `pubkey` is the same as the parameter used when registering
+     */   
+    function deregisterOperator(address operator, uint8[] memory quorumNumbers, BN254.G1Point memory pubkey) external onlyRegistryCoordinator returns(bytes32){
```

Comment:
The quorum numbers will be formatted as such: https://github.com/Layr-Labs/eigenlayer-contracts/blob/1fb084c35454e8fb4a453ab6a2e529667bd36584/src/contracts/middleware/StakeRegistry.sol#L299

Reply:
done